backend/pkg/transport/network/tcp/tcp_test.go

```diff
@@ -0,0 +1,86 @@
+package tcp_test
+
+import (
+	"encoding/json"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/network/tcp"
+	"github.com/rs/zerolog"
+	"os"
+	"testing"
+	"time"
+)
+
+type Socket struct {
+	SrcIP   string
+	SrcPort uint16
+	DstIP   string
+	DstPort uint16
+}
+
+// Payload defines a piece of information comming from the network with its metadata
+type Payload struct {
+	Timestamp time.Time
+	Socket    Socket
+	Data      []byte
+}
+
+func TestTCP(t *testing.T) {
+	writer := os.Stdout
+	logger := zerolog.New(writer)
+
+	// Server setup
+	serverAddr := "127.0.0.1:3000"
+	server := tcp.NewServer(serverAddr, tcp.ServerConfig{}, logger)
+	go func() {
+		if err := server.Listen(); err != nil {
+			t.Fatalf("Server failed to listen: %v", err)
+		}
+	}()
+	time.Sleep(1 * time.Second)
+
+	// Client setup
+	client := tcp.NewClient(serverAddr, tcp.ClientConfig{}, logger)
+	conn, err := client.Dial()
+	if err != nil {
+		t.Fatalf("Client failed to dial: %v", err)
+	}
+	defer conn.Close()
+
+	// Create and send Payload
+	originalPayload := Payload{
+		Timestamp: time.Now(),
+		Socket: Socket{
+			SrcIP:   "127.0.0.1",
+			SrcPort: 3000,
+			DstIP:   "127.0.0.1",
+			DstPort: 3000,
+		},
+		Data: []byte("hello"),
+	}
+	payloadBytes, err := json.Marshal(originalPayload)
+	if err != nil {
+		t.Fatalf("Failed to serialize Payload: %v", err)
+	}
+	_, err = conn.Write(payloadBytes)
+	if err != nil {
+		t.Fatalf("Failed to write Payload to server: %v", err)
+	}
+
+	// Read and verify Payload
+	buffer := make([]byte, 1024)
+	n, err := conn.Read(buffer)
+	if err != nil {
+		t.Fatalf("Failed to read from server: %v", err)
+	}
```

Comment:
the payload just sent should be received by the server, not the same client. This will never end. Maybe in the goroutine above for the server listen make it so that the server reads, checks the read is correct and then writes. That way you test both ways of the communication